crates/bevy_render/src/view/mod.rs

```diff
@@ -809,24 +842,23 @@ pub fn prepare_view_targets(
         &CameraMainTextureUsages,
         &Msaa,
     )>,
-    manual_texture_views: Res<ManualTextureViews>,
+    view_target_textures: Res<ViewTargetTextures>,
 ) {
     let mut textures = HashMap::default();
-    let mut output_textures = HashMap::default();
     for (entity, camera, view, texture_usage, msaa) in cameras.iter() {
         let (Some(target_size), Some(target)) = (camera.physical_target_size, &camera.target)
         else {
             continue;
         };
 
-        let Some(out_texture) = output_textures.entry(target.clone()).or_insert_with(|| {
-            target
-                .get_texture_view(&windows, &images, &manual_texture_views)
-                .zip(target.get_texture_format(&windows, &images, &manual_texture_views))
-                .map(|(view, format)| {
-                    OutputColorAttachment::new(view.clone(), format.add_srgb_suffix())
-                })
-        }) else {
+        let Some(out_texture) = view_target_textures
+            .get(target)
+            .expect("Something went wrong with prepare_view_textures")
+            .as_ref()
+            .map(|(view, format)| {
+                OutputColorAttachment::new(view.clone(), format.add_srgb_suffix())
+            })
```

Comment:
I think there may be a subtle logic difference here. In the original, it looks if multiple cameras which render to the same target all use the same attachment. In the new version, every camera gets it's own attachment. At this point I'm not sure if that's going to effect how things work.

Reply:
Cleaning this up actually simplified the code even more. Also added docs!